### PR TITLE
Pass multipart form fields to upload endpoint

### PR DESCRIPTION
**Pull request by goto-bus-stop**
Send the custom form fields defined by the Uppy client along to the
upload endpoint. Allows eg. the AwsS3 plugin to work for remote files.

Fixes https://github.com/transloadit/uppy-server/issues/57

Tested using files from Google Drive & uploading to S3 using the AwsS3 plugin.

### CODE REVIEW

**Comment by ifedapoolarewaju** [comment]
> Tested using files from Google Drive & uploading to S3 using the AwsS3 plugin.

@goto-bus-stop quick question about this, so for this to work, the request would not be signed right? So the aws `accessKeyId` (and maybe the secret) would be sent directly from the frontend instead?

**Comment by goto-bus-stop** [comment]
The request to S3 will be signed. The signature is sent along as a form field. The flow would look like this:

 - Client gets information about remote file from server
 - Client gets signed upload parameters from `$uppy-server/s3/params?filename&type`
 - Client tells server to upload the remote file to the S3 URL with some form fields (the ones it would otherwise use to upload to S3 directly)

S3 doesn't check where the fields come from, they must just be signed correctly, which already happened in the previous step. So whether it's from uppy-server or a browser, both work. With uploads using a presigned PUT URL (so parameters that were not generated by uppy-server), it will still work, because the signature is in the URL query string.

Basically this is still working exactly the same way, the client still doesn't know anything about the access key or the secret key. The only difference between local and remote upload is that it's going to originate from a different machine.

**Review by ifedapoolarewaju — APPROVED** [review]
looks good to me, thanks! 🙂
